sda/cmd/ingest/ingest.go

```diff
@@ -137,440 +149,445 @@ func main() {
+					log.Errorf("Failed to Nack message, reason: (%s)", err.Error())
 				}
-
-				switch status {
-				case "disabled":
-					fileID, err := db.GetFileID(delivered.CorrelationId)
-					if err != nil {
-						log.Errorf("failed to get ID for file, reason: %s", err.Error())
-						if err := delivered.Nack(false, true); err != nil {
-							log.Errorf("failed to Nack message, reason: (%s)", err.Error())
-						}
-
-						continue
-					}
-
-					fileInfo, err := db.GetFileInfo(fileID)
-					if err != nil {
-						log.Errorf("failed to get info for file: %s", fileID)
-						if err := delivered.Nack(false, true); err != nil {
-							log.Errorf("failed to Nack message, reason: (%s)", err.Error())
-						}
-
-						continue
-					}
-
-					if err = db.UpdateFileEventLog(fileInfo.Path, "enabled", delivered.CorrelationId, "ingest", "{}", string(delivered.Body)); err != nil {
-						log.Errorf("failed to set ingestion status for file from message: %v", delivered.CorrelationId)
-						if err := delivered.Nack(false, true); err != nil {
-							log.Errorf("failed to Nack message, reason: (%s)", err.Error())
-						}
-
-						continue
-					}
-
-					if fileInfo.Checksum != "" {
-						msg := schema.IngestionVerification{
-							User:        message.User,
-							FilePath:    message.FilePath,
-							FileID:      fileID,
-							ArchivePath: fileInfo.Path,
-							EncryptedChecksums: []schema.Checksums{
-								{Type: "sha256", Value: fileInfo.Checksum},
-							},
-						}
-						archivedMsg, _ := json.Marshal(&msg)
-						err = schema.ValidateJSON(fmt.Sprintf("%s/ingestion-verification.json", conf.Broker.SchemasPath), archivedMsg)
-						if err != nil {
-							log.Errorf("Validation of outgoing message failed, reason: (%s)", err.Error())
-
-							continue
-						}
-						if err := mq.SendMessage(delivered.CorrelationId, conf.Broker.Exchange, conf.Broker.RoutingKey, archivedMsg); err != nil {
-							log.Errorf("failed to publish message, reason: (%s)", err.Error())
-
-							continue
-						}
-
-						if err := delivered.Ack(false); err != nil {
-							log.Errorf("failed to Ack message, reason: (%s)", err.Error())
-						}
-
-						continue
-					}
-				case "":
-					// Catch all for inboxes that doesn't update the DB
-					fileID, err = db.RegisterFile(message.FilePath, message.User)
-					if err != nil {
-						log.Errorf("InsertFile failed, reason: (%s)", err.Error())
-						if err := delivered.Nack(false, true); err != nil {
-							log.Errorf("failed to Nack message, reason: (%s)", err.Error())
-						}
-
-						continue
-					}
-				default:
-					fileID, err = db.GetFileID(delivered.CorrelationId)
-					if err != nil {
-						log.Errorf("failed to get ID for file, reason: %s", err.Error())
-						if err := delivered.Nack(false, true); err != nil {
-							log.Errorf("failed to Nack message, reason: (%s)", err.Error())
-						}
-
-						continue
-					}
+			default:
+				log.Errorln("better message needed")
```

Comment:
Will it be better to print the content of the message? Otherwise it is not easy to debug why it is failed. Also, it is not very clear what does `better message` mean.

Reply:
Did I forget to change that? It should hold a relevant message.